pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -924,11 +924,10 @@ func (vs *VSphere) DiskIsAttached(volPath string, nodeName k8stypes.NodeName) (b
 	}
 
 	if !nodeExist {
-		glog.Warningf(
-			"Node %q does not exist. DiskIsAttached will assume vmdk %q is not attached to it.",
-			vSphereInstance,
-			volPath)
-		return false, nil
+		glog.Errorf("Node %q does not exist. DiskIsAttached will throw an error as node doesn't exist.",
+			vSphereInstance)
+		return false, errors.New(fmt.Sprintf("Node %q does not exist. DiskIsAttached will throw an error as node doesn't exist.",
```

Comment:
the error log could be " DiskIsAttached failed to determine whether disk is still attached: node %q does not exist.